lib/src/revset.rs

```diff
@@ -755,6 +775,22 @@ pub fn expect_string_pattern(node: &ExpressionNode) -> Result<StringPattern, Rev
     revset_parser::expect_pattern_with("string pattern", node, parse_pattern)
 }
 
+fn expect_timeexpression<Tz: TimeZone>(
+    node: &ExpressionNode,
+    now: DateTime<Tz>,
+) -> Result<TimeExpression, RevsetParseError>
+where
+    Tz::Offset: Copy,
+{
+    let expression = revset_parser::expect_literal::<String>("time expression", node)?;
+    TimeExpression::parse(expression.as_str(), now).map_err(|err| {
+        RevsetParseError::expression(
+            format!("Unable to parse time expression: #{err}"),
```

Comment:
nit: bad `#` char?

The error can also be attached as `.with_source(err)`.

Reply:
`#` removed, added `.with_source(err)`.